src/routing/tests/mod.rs

```diff
@@ -102,6 +110,47 @@ async fn receive_join_request() -> Result<()> {
         .await?
         .into_iter();
 
+    let response_message = assert_matches!(
+        commands.next(),
+        Some(Command::SendMessage { message, .. }) => message
+    );
+    let response_message = Message::from_bytes(&response_message)?;
+
+    let (nonce, signature) = assert_matches!(
+        response_message.variant(),
+        Variant::Challenge { nonce, signature, .. } => (*nonce, *signature)
+    );
+
```

Comment:
I'd prefer to stop this test here and to add a separate test for the following logic. So one test for receiving join request without resource proof and another one for with resource proof. Would keep the pattern where we handle only one command per test which makes it simpler to read. 

We should also consider splitting this file into multiple ones, as it has become quite big. Can be done in a different PR though.